### PR TITLE
Some enhancements

### PR DESCRIPTION
**Pull request by xepozz**
- [X] Set horizontal margin 1px on each side. The condition was always `true`
- [X] Change `is` return type to `boolean`
- [X] A few cosmetic changes

### CODE REVIEW

File: src/components/DataTypes/Object.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const PreObjectType: React.FC<DataItemProps<object>> = (props) => {
             <CircularArrowsIcon sx={{
               fontSize: 12,
               color: textColor,
-              mx: sizeOfValue ? 0.5 : 0
+              mx: 1
```
Comment (Himself65) [inline src/components/DataTypes/Object.tsx:68]: does this cause the icon too wide? I'm not a UI designer not sure which is better

Screenshot from 2022-09-22 19-37-20

File: src/utils/index.ts [hunk 2]
```diff
@@ -25,52 +25,52 @@ export const applyValue = (obj: any, path: (string | number)[], value: any) => {
 
 // case 1: you only render with a single component
 export function createDataType<ValueType = unknown> (
-  is: (value: unknown) => boolean,
+  is: (value: ValueType) => boolean,
```
Comment (Himself65) [inline src/utils/index.ts:28]: I think we don't guarantee the value is `ValueType.` value must be one of `any` or `unknown`

File: src/components/DataTypes/Object.tsx (hunk 1 above)
Comment (xepozz) [reply src/components/DataTypes/Object.tsx]: Decreased to `0.5` that means 4px on each side (I hope :))

Comment (xepozz) [reply src/components/DataTypes/Object.tsx]: Yeap. That's better.

**Review by Himself65 — APPROVED** [review]

**Pull request merged**